test/parallel/test-no-enter-tickcallback.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+var allsGood = false;
+var cntr = 0;
+
+process.on('exit', () => {
+  assert.ok(cntr > 0, '_tickDomainCallback was never called');
+});
+
+/**
+ * This test relies upon the following internals to work as specified:
+ *  - require('domain') causes node::Environment::set_tick_callback_function()
+ *    to use process._tickDomainCallback() to process the nextTickQueue;
+ *    replacing process._tickCallback().
+ *  - setImmediate() uses node::MakeCallback() instead of
+ *    node::AsyncWrap::MakeCallback(). Otherwise the test will always pass.
+ *    Have not found a way to verify that node::MakeCallback() is used.
+ */
+process._tickDomainCallback = function _tickDomainCallback() {
+  assert.ok(allsGood, '_tickDomainCallback should not have been called');
+  cntr++;
+};
+
+setImmediate(() => {
+  require('domain');
+  setImmediate(() => setImmediate(() => {
+    allsGood = true;
+    process.nextTick(() => {});
```

Comment:
I can set `mustCall()` around the `setImmediate()`'s, but not here. Since by overriding `_tickDomainCallback()` we loose the ability to process the `nextTickQueue`.